scope/xgb.py

```diff
@@ -0,0 +1,107 @@
+import datetime
+import os
+import pathlib
+from .models import AbstractClassifier
+
+#!git clone https://github.com/DTrimarchi10/confusion_matrix.git
+import numpy as np
+from xgboost import XGBClassifier
+import xgboost as xgb
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score, confusion_matrix, matthews_corrcoef, mean_absolute_error
+from sklearn.metrics import f1_score, ConfusionMatrixDisplay
+#plot_confusion_matrix, is deprecated, so import ConfusionMatrixDisplay instead
+from sklearn.metrics import roc_curve, auc, precision_recall_curve, classification_report, average_precision_score
+import pandas as pd
+
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+import os
+import sys
+import copy
+import json
+
+    
+
+class XGB(AbstractClassifier):
+    """Baseline model with a statically-defined architecture"""
+    
+    def setup(
+        self,
+        'max_depth':6,
+        'min_child_weight': 1,
+        'eta':.1,
+        'subsample': 0.7,
+        'colsample_bytree': 0.7,
+        'objective': 'binary:logistic',
+        'eval_metric': 'auc',
+        'early_stopping_rounds': 10,
+        'num_boost_round': 999,
+        X_train,
+        X_test,
+        
+        **kwargs,
+    ):
+      #  removed for now:
+      #  'gpu_id': 0,
+      #  'tree_method': 'gpu_hist',
+
+    evals = {'dtrain': X_train, 'dtest': X_test}
+
+    params = {'max_depth': max_depth,
+    'min_child_weight': min_child_weight,
+    'eta': eta,
+    'subsample': subsample,
+    'colsample_bytree': colsample_bytree,
+    'objective': objective,
+  #  'gpu_id': gpu_id,
+  #  'tree_method': tree_method,
+    'eval_metric': eval_metric,
+    'early_stopping_rounds': early_stopping_rounds,
+    'num_boost_round': num_boost_round
+    }
+    
+    self.meta['params'] = params
+    self.meta['evals'] = evals
+
+
+    def train(self,X_train, X_test, **kwargs):
+
+        self.model = xgb.train(self,params,evals, **kwargs)
+        
+        final_train = xgb.DMatrix(dtrain, label=y_train)
+        final_test = xgb.DMatrix(dtest, label=y_test)
```

Comment:
We may want to use `DMatrix` before calling `xgb.train` to get the input data in the right format. I'm currently reviewing how `scope.py` manages training/validation/test datasets to integrate XGB with the current code smoothly.